tests/BenchmarkDotNet.IntegrationTests/InProcessEmitTest.cs

```diff
@@ -239,35 +239,125 @@ public static ValueTask<decimal> InvokeOnceStaticValueTaskOfT()
             }
         }
 
-        [Fact]
-        public void InProcessEmitToolchainSupportsIterationSetupAndCleanup()
+        [Theory]
+        [InlineData(typeof(IterationSetupCleanup))]
+        [InlineData(typeof(GlobalSetupCleanupTask))]
+        [InlineData(typeof(GlobalSetupCleanupValueTask))]
+        [InlineData(typeof(GlobalSetupCleanupValueTaskSource))]
+        public void InProcessEmitToolchainSupportsSetupAndCleanup(Type benchmarkType)
```

Comment:
So far this test was a `[Fact]` (having a single test case), so we did not have to worry about multiple tests accessing the same static field at the same time.

As long as we keep the test parallelization disabled here:
https://github.com/dotnet/BenchmarkDotNet/blob/c7ed714a34a4de04f7320b6d971a51b04545bb1c/tests/BenchmarkDotNet.IntegrationTests/xunit.runner.json#L6-L7

It's OK. But if we ever change our policy on that, multiple tests may update `Counters` fields at the same time and make this test flaky. 

There is no need to change anything as of now,  just something to keep in mind when writing more tests in the future.

Reply:
According to the [documentation](https://xunit.net/docs/running-tests-in-parallel.html),

> By default, each test class is a unique test collection. Tests within the same test class will not run in parallel against each other.

So we should be good, regardless of that setting.